examples/api/lib/material/selection_area/selection_area.2.dart

```diff
@@ -0,0 +1,354 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math';
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+
+/// Flutter code sample for [SelectionArea].
+
+void main() => runApp(const SelectionAreaColorTextRedExampleApp());
+
+class SelectionAreaColorTextRedExampleApp extends StatelessWidget {
+  const SelectionAreaColorTextRedExampleApp({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      title: 'Flutter Demo',
+      theme: ThemeData(
+        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
+        useMaterial3: true,
+      ),
+      home: const MyHomePage(title: 'Flutter Demo Home Page'),
+    );
+  }
+}
+
+class MyHomePage extends StatefulWidget {
+  const MyHomePage({super.key, required this.title});
+
+  final String title;
+
+  @override
+  State<MyHomePage> createState() => _MyHomePageState();
+}
+
+typedef _LocalSpanRange = ({int startOffset, int endOffset});
+
+class _MyHomePageState extends State<MyHomePage> {
+  final SelectionListenerNotifier _selectionNotifier = SelectionListenerNotifier();
+  final ContextMenuController _menuController = ContextMenuController();
+  final GlobalKey<SelectionAreaState> selectionAreaKey = GlobalKey<SelectionAreaState>();
+
+  // The data of the top level TextSpans. Each TextSpan is mapped to a _LocalSpanRange,
+  // which is the range the textspan covers relative to the SelectionListener it is under.
+  Map<_LocalSpanRange, TextSpan> dataSourceMap = <_LocalSpanRange, TextSpan>{};
+  // The data of the bulleted list contained within a WidgetSpan. Each bullet is mapped
+  // to a _LocalSpanRange, being the range the bullet covers relative to the SelectionListener
+  // it is under.
+  Map<_LocalSpanRange, TextSpan> bulletSourceMap = <_LocalSpanRange, TextSpan>{};
+  Map<int, Map<_LocalSpanRange, TextSpan>> widgetSpanMaps = <int, Map<_LocalSpanRange, TextSpan>>{};
+  // The origin data used to restore the demo to its initial state.
+  late final Map<_LocalSpanRange, TextSpan> originSourceData;
+  late final Map<_LocalSpanRange, TextSpan> originBulletSourceData;
+
+  @override
+  void initState() {
+    super.initState();
+    _initData();
+    _selectionNotifier.addStatusListener(_handleSelectionStatusChanged);
+  }
+
+  @override
+  void dispose() {
+    _selectionNotifier.removeStatusListener(_handleSelectionStatusChanged);
+    _selectionNotifier.dispose();
+    super.dispose();
+  }
+
+  void _initData() {
```

Comment:
Nit: I would put these private methods above the `@override` methods initState and dispose. Maybe that's just me though. I thought I was following the styleguide but maybe it's not totally clear on this. https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#order-other-class-members-in-a-way-that-makes-sense

Reply:
I see your point, I always put private methods in between my `build` method and other `state` overrides, but I think this makes more sense.